### PR TITLE
feat(rpc): support more chromium-specific apis

### PR DESCRIPTION
**Pull request by dgozman**
This includes page CDPSession, backgroundPages() and serviceWorkers().

This has also revealed an issue with closing order between the context and the service worker.

### CODE REVIEW

File: src/rpc/client/browserContext.ts [hunk 1]
```diff
@@ -26,9 +26,14 @@ import { Browser } from './browser';
 import { ConnectionScope } from './connection';
 import { Events } from '../../events';
 import { TimeoutSettings } from '../../timeoutSettings';
+import { CDPSession } from './cdpSession';
+import { Events as ChromiumEvents } from '../../chromium/events';
+import { Worker } from './worker';
 
 export class BrowserContext extends ChannelOwner<BrowserContextChannel, BrowserContextInitializer> {
   _pages = new Set<Page>();
+  _backgroundPages = new Set<Page>();  // Chromium-specific.
```
Comment (pavelfeldman) [inline src/rpc/client/browserContext.ts:35]: Prefix cr instead of comments?

File: src/rpc/server/browserContextDispatcher.ts [hunk 2]
```diff
@@ -18,17 +18,32 @@ import * as types from '../../types';
...
+import { CDPSessionDispatcher } from './cdpSessionDispatcher';
+import { Events as ChromiumEvents } from '../../chromium/events';
 
 export class BrowserContextDispatcher extends Dispatcher<BrowserContext, BrowserContextInitializer> implements BrowserContextChannel {
   private _context: BrowserContextBase;
 
   constructor(scope: DispatcherScope, context: BrowserContextBase) {
+    // Chromium-specific.
+    let backgroundPages: PageDispatcher[] = [];
+    if ((context as any).backgroundPages)
```
Comment (pavelfeldman) [inline src/rpc/server/browserContextDispatcher.ts:35]: contextbase has a browser base that has a browser type that has a name :)

**Review by pavelfeldman — APPROVED** [review]

File: src/rpc/client/browserContext.ts (hunk 1 above)
Comment (dgozman) [reply src/rpc/client/browserContext.ts]: Done.

File: src/rpc/server/browserContextDispatcher.ts (hunk 2 above)
Comment (dgozman) [reply src/rpc/server/browserContextDispatcher.ts]: Plumbed.

**Pull request merged**